enhancements/security/bmc-ca-certificate-support.md

```diff
@@ -0,0 +1,278 @@
+
+## Design Details
+
+### Test Plan
+- Unit Test
+  - Add UT for CBO.
+
+- E2E test
+  - Configure a Redfish job to interact with Ironic.
+  - Set up sushy-tools to use a self-signed certificate, ensuring it is configured to work
+    alongside the Redfish job.
+  - Provide the bmcCACert to ironic and ensure that certificate validation is enable.
+  - Execute the end-to-end test to verify if the cluster deployment is successful.
+    Successful deployment would indirectly suggest that the certificate configuration is
+    correctly recognized and utilized by Ironic.
+
+### Graduation Criteria
+
+This work will be GA immediately since there is no phasing possible or planned.
+
+#### Dev Preview -> Tech Preview
+
+N/A
+
+#### Tech Preview -> GA
+
+N/A
+
+#### Removing a deprecated feature
+
+N/A
+
+### Upgrade / Downgrade Strategy
+
+This enhancement should not affect the Upgrade/Downgrade Strategy.
+
+### Version Skew Strategy
+
+This enhancement requires the support of both `cluster-baremetal-operator`
+and `ironic-image`. The version of ironic-image is controlled by
+`cluster-baremetal-operator`, thus there will be no version skew.
+
+### Operational Aspects of API Extensions
+
+N/A
+
+#### Failure Modes
+
+1. Before the IPI installation begins, the `installer` will perform a
+   pre-validation using the provided CA certificate. If the validation fails,
+   an error will be reported immediately, indicating which baremetal BMC failed
+   the validation, and the installation will not proceed.
+
+2. If the pre-validation passes, but a certificate validation error occurs
+   during the installation process (e.g., due to the changes to the baremetal
+   BMC certificate or a man-in-the-middle attack), the user will not be able to
+   see the error message directly until timeout. However, if the user notices
+   that the installation duration is abnormal, they can check the `ironic`
+   container for error messages similar to the following:
+
+   ```log
+   ERROR ironic.conductor.utils requests.exceptions.SSLError: HTTPSConnectionPool(host='xxx.xxx.xxx.xxx', port=443): Max retries exceeded with url: /rest/v1/Oem/eLCM/ProfileManagement/BiosConfig (Caused by SSLError(SSLError(524297, '[X509] PEM lib (_ssl.c:4293)')))
+   ```
+
+#### Support Procedures
+
+TBD
+
+## Implementation History
+
+TBD
+
+## Alternatives
+
+An alternative approach would be to add these CA certificates through the `additionalTrustBundle`.
+However, this has been discussed in a similar [enhancement][OpenStack enhancement]
+and was deemed not to be the optimal way.
+For our enhancement, there are two primary reasons to consider:
+1. It cannot be applied to the BMC certificate verification during control plane creation.
+2. There is also the `additionalTrustBundlePolicy` that can be set to either Always or Proxyonly,
+  in the latter case using it for the BMCs would not make sense (and this would leave us without
+  another way to set the trust bundle for the BMCs).
```

Comment:
I missed this while making (https://github.com/openshift/enhancements/pull/1395#issuecomment-1739494282). The rationale behind the openstack is enhancement is the second point I was trying to make: if you want to be more selective about where the trust is added, then this approach makes sense. 

I'm not sure I follow either of the bullet points here, though. For 1, it's unclear why `additionalTrustBundle` would have this limitation. For 2, I'm not sure why Proxyonly is an issue. If the issue is you want to have separate CA for proxies, like in openstack, then this should be slightly reworded.